components/locale/src/langid.rs

```diff
@@ -62,6 +62,9 @@ pub struct LanguageIdentifier {
     pub variants: subtags::Variants,
 }
 
+#[cfg(feature = "use-ecma402")]
+impl ecma402_traits::Locale for LanguageIdentifier {}
```

Comment:
Why do you need this here? You can implement it in `ecma402_traits`, no? Or do you envision `ecma402_traits` not depending on `icu4x` at all?

(In my mind it was `ecma402_traits` that had a feature called `icu4x` :))

Reply:
ecma402_traits should IMHO be exactly that - trait definitions - without an implementation.

Reply:
To wit, ecma402 implementation is split across two parts: traits only (and minimum configuration per spec), and the implementation proper.

I wanted to avoid needing to implement a custom adaptor for `ecma402_traits::Locale`, which necessitates this approach.  The idea would be that `LanguageIdentifier` would be a drop-in replacement for whatever ICU uses to represent locale.

But I suppose I could wrap LanguageIdentifier into a Locale and work from there.

Reply:
I pulled the implementation of `ecma402_traits::Locale` into the icu4x-specific implementation crate, which allowed me to remove the dependency of icu4x on ecma402 traits.  I should probably do that with rust_icu as well.